pkg/interop/native/ledger/transaction_signer.go

```diff
@@ -0,0 +1,65 @@
+package ledger
+
+import "github.com/nspcc-dev/neo-go/pkg/interop"
+
+// TransactionSigner represent the signer of a NEO transaction. It's similar to
+// Signer class in Neo .net framework.
+type TransactionSigner struct {
+	// Bytes represents the binary serialized form of the given signer.
+	Bytes []byte
+	// Account represents the account (160 bit BE value in a 20 byte slice) of
+	// the given signer.
+	Account interop.Hash160
+	// Scopes represents the set of witness scopes of the given signer.
+	Scopes SignerScope
+	// Contracts represents the set of contract hashes (160 bit BE value in a 20
+	// byte slice) allowed to be called by the signer.
+	AllowedContracts []interop.Hash160
+	// AllowedGroups represents the set of contract groups (ecdsa public key
+	// bytes in a 32 byte slice) allowed to be called by the signer.
+	AllowedGroups []interop.PublicKey
+	// Rules represents a rule-based witness scope of the given signer.
+	Rules []WitnessRule
+}
+
+// SignerScope represents a signer's witness scope.
+type SignerScope byte
+
+// Various witness scopes.
+const (
+	// None specifies that no contract was witnessed. Only sign the transaction.
+	None SignerScope = 0
+	// CalledByEntry means that this condition must hold:
+	// EntryScriptHash == CallingScriptHash. No parameters are needed, as the
+	// witness/permission/signature given on first invocation will automatically
+	// expire if entering deeper internal invokes. This can be default safe
+	// choice for native NEO/GAS.
```

Comment:
We're looking at it from the contract's perspective, so we're not trying to choose a scope, rather deciphering what is used by the transaction.